File: two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        complements = {}

        for i in range(len(nums)):
            if nums[i] in complements:
                return [complements[nums[i]], i]
            else:
                complements[target - nums[i]] = i

        raise Exception('Something is wrong!')


if __name__ == '__main__':
    s = Solution()
    nums = [2, 7, 11, 15]
    target = 9

    print(s.twoSum(nums, target))

File: two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  var i;
  var complements = {};

  for (i = 0; i < nums.length; i++) {
    if (nums[i] in complements) {
      return [complements[nums[i]], i];
    } else {
      complements[target - nums[i]] = i;
    }
  }

  throw 'Something is wrong';
};

var main = function() {
  var nums = [2, 7, 11, 15];
  var target = 9;

  console.log(twoSum(nums, target));
};

if (require.main === module) {
  main();
}
